src/components/map/components/mobility/GeofencingZones.tsx

```diff
@@ -0,0 +1,93 @@
+import MapboxGL from '@rnmapbox/maps';
+
+import {FeatureCollection} from 'geojson';
+
+import {
+  PreProcessedGeofencingZones,
+  usePreProcessedGeofencingZones,
+} from '@atb/components/map';
+import {useVehicleQuery} from '@atb/mobility/queries/use-vehicle-query';
+
+import {hitboxCoveringIconOnly} from '@atb/components/map';
+import {VehicleExtendedFragment} from '@atb/api/types/generated/fragments/vehicles';
+
+type GeofencingZonesProps = {
+  selectedVehicleId: string;
+};
+export const GeofencingZones = ({selectedVehicleId}: GeofencingZonesProps) => {
+  const {
+    data: vehicle,
+    isLoading,
+    isError,
+  } = useVehicleQuery(selectedVehicleId);
+
+  if (!vehicle || !selectedVehicleId || isLoading || isError) {
+    return <></>;
+  }
+  return <GeofencingZonesForVehicle vehicle={vehicle} />;
```

Comment:
Just a FYI about react-query: When using `isSuccess`, then you also get type safety that `data` is present. So the check `!vehicle || !selectedVehicleId || isLoading || isError` is a little more complicated than it needs to be.

I think this would do it:
```tsx
const vehiclesQuery = useVehicleQuery(selectedVehicleId);

return vehiclesQuery.isSuccess
  ? <GeofencingZonesForVehicle vehicle={vehiclesQuery.data} />
  : null;
```

Reply:
You would think so, but this still gives "Type 'VehicleExtendedFragment | undefined' is not assignable to type 'VehicleExtendedFragment'."

Reply:
Then the success type of `useVehicleQuery` can be `undefined`, which is [not allowed by react-query](https://tanstack.com/query/v4/docs/framework/react/guides/migrating-to-react-query-4#undefined-is-an-illegal-cache-value-for-successful-queries) and will give runtime error. You can take a quick look into that.